### PR TITLE
Clean-up Parser and Lexer

### PR DESCRIPTION
**Pull request by wenovus**
- Remove unused exported function `FakeStatement`.
- Simplify by replacing `closeBrace` and `openBrace` with their character values.
- Rename `tIdentifier` to `tUnquoted`. Identifiers can be quoted in YANG, so this name is confusing. The original author intended these to be unquoted strings.
- Improve comments in the parser.

### CODE REVIEW

File: pkg/yang/parse.go [hunk 1]
```diff
@@ -162,8 +151,8 @@ func (s *Statement) Write(w io.Writer, indent string) error {
 	return nil
 }
 
-// ignoreMe is returned to continue processing after an error (the parse will
-// fail, but we want to look for more errors).
+// ignoreMe is an error recovery non-terminal used by the parser in
```
Comment (robshakir) [inline pkg/yang/parse.go:154]: "error recovery non-terminal"? Should we say here "a non-terminal error that can be used in order...".

File: pkg/yang/parse.go [hunk 2]
```diff
@@ -185,19 +174,18 @@ Loop:
...
 		case p.hitBrace:
-			fmt.Fprintf(p.errout, "%s:%d:%d: unexpected %c\n", ns.file, ns.line, ns.col, closeBrace)
+			fmt.Fprintf(p.errout, "%s:%d:%d: unexpected %c\n", ns.file, ns.line, ns.col, '}')
 		default:
 			statements = append(statements, ns)
 		}
 	}
 
-	p.checkStatementDepth()
+	p.checkStatementDepthIsZero()
```
Comment (robshakir) [inline pkg/yang/parse.go:183]: Do we need `check` in the name here? `statementDepthIsZero`?

**Review by robshakir — CHANGES_REQUESTED** [review]

File: pkg/yang/parse.go (hunk 1 above)
Comment (wenovus) [reply pkg/yang/parse.go]: Changed to "token" to describe this.

I think it's in the same spirit as the "error terminal" in the bison parser (last paragraph): https://www.gnu.org/software/bison/manual/html_node/Symbols.html

File: pkg/yang/parse.go (hunk 2 above)
Comment (wenovus) [reply pkg/yang/parse.go:183]: I think the `check` is descriptive because it can output an error if the check fails.

**Review by robshakir — APPROVED** [review]

**Pull request merged**